src/frontend/src/routes/(new-styling)/+page.svelte

```diff
@@ -78,12 +79,21 @@
             <ul class="contents">
               {#each lastUsedIdentities as identity}
                 <li class="contents">
-                  <ButtonCard onclick={() => handleContinueAs(identity)}>
+                  <ButtonCard
+                    onclick={() => handleContinueAs(identity)}
+                    disabled={nonNullish(
+                      authLastUsedFlow.authenticatingIdentity,
+                    )}
+                  >
                     <Avatar size="sm">
-                      <UserIcon size="1.25rem" />
+                      {#if identity.identityNumber === authLastUsedFlow.authenticatingIdentity}
+                        <ProgressRing />
```

Comment:
Putting it inside the Avatar avoids layout shifts and it looks kind of nice also!